dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/header/HeaderExchangeClient.java

```diff
@@ -55,16 +54,16 @@ public HeaderExchangeClient(Client client, boolean needHeartbeat) {
 
         this.heartbeat = client.getUrl().getParameter(Constants.HEARTBEAT_KEY, dubbo != null &&
                 dubbo.startsWith("1.0.") ? Constants.DEFAULT_HEARTBEAT : 0);

```

Comment:
`private final int NO_HEARTBEAT = 0;`
can we replace **0** in case if **dubbo** does not start with **1.0** with NO_HEARTBEAT for better readability.

Reply:
> > @kexianjun In my opinion, idle check timer is responsible to a `Connection`, not a `Service`.
> 
> 目前的处理方式，一个HashedWheelTimer最多只处理两个任务(server端是一个，clien端是两个)，每个HashedWheelTimer都会启动一个线程检测，如果一个服务作为消费方调用服务比较多的情况下，就需要起很多线程做心跳检查，比如说我们的一个业务系统，调用其他不同系统的服务多达上十个，再加上线上集群部署，单是心跳检测线程就要起二三十个了。我觉得检测的任务和连接是对应的，定时器是可以共享的。

Got it. I will make this change.

Reply:
The magic number is indeed troublesome, but it isn't related to this pull request. Let us change them in other pr.